src/index.js

```diff
@@ -0,0 +1,177 @@
+
+/**********************1*********************/
+
+
+
+
+const data = [
+  { value: "1", label: "One" },
+  { value: "2", label: "Two" },
+  { value: "3", label: "Three" },
+];
+
+const values = data.reduce((values, { value }) => {
+  values.push(value);
+  return values;
+}, []);
+
+
+
+/**********************2*********************/
+
+
+async function getIndexes() {
+  return await fetch('https://api.coingecko.com/api/v3/indexes').then(res => res.json());
+}
+
+async function analyzeIndexes() {
+  const indexes = await getIndexes().catch(_ => {
+    throw new Error('Unable to fetch indexes');
+  });
+  return indexes;
+}
+
+
+
+/**********************3*********************/
+
+
+
+let state;
+const user = getUser();
+if (user) {
```

Comment:
the JS ternary conditional operator can save you a lot of code here
 
ctx.body = {user : getUser() || null}
ctx.body.project = ctx.body.user ? getProject(ctx.body.user.id) : null;